frontend/catalyst/passes.py

```diff
@@ -52,7 +52,21 @@ def __init__(self, name, *options, **valued_options):
         self.valued_options = valued_options
 
     def __repr__(self):
-        return self.name + ", ".join(self.options)
+        return (
+            self.name
+            + " ".join(f"--{option}" for option in self.options)
+            + " ".join(f"--{option}={value}" for option, value in self.valued_options)
+        )
+
+
+class PassPlugin(Pass):
+    """Class intended to hold options for pass plugins"""
+
+    def __init__(self, path, name, *options, **valued_options):
+        assert EvaluationContext.is_tracing()
+        EvaluationContext.add_plugin(path)
+        self.path = path
```

Comment:
Does any type-checking need to be performed on `path` here? e.g. does it need to be a `pathlib.Path`, or can it be a string as well? A type hint would also be useful :)

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1317/commits/a702987fb433f15569c6d1ce296f596e789b8bbb